### PR TITLE
Move env vars to docker compose

### PR DESCRIPTION
**Pull request by srstsavage**
Move env vars from config.sh to docker-compose.yml for visibility and simplicity (leaving a note about `config.sh` in docker-compose.yml for an alternate approach).

Also remove `version: 2` from docker-compose.yml to use the modern [Compose Specification](https://compose-spec.io/)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,18 +1,38 @@
...
+      ERDDAP_emailDailyReportsTo:
+      ERDDAP_emailFromAddress:
+      ERDDAP_emailUserName:
+      ERDDAP_emailPassword:
+      ERDDAP_emailProperties:
+      ERDDAP_emailSmtpHost:
+      ERDDAP_emailSmtpPort:
+      ERDDAP_adminInstitution: Integrated Ocean Observing System (IOOS)
+      ERDDAP_adminInstitutionUrl: https://ioos.github.io/
+      ERDDAP_adminIndividualName: Micah Wengren
```
Comment (mwengren) [inline docker-compose.yml:30]: I'm not sure how my contact info made it in here originally, but let's please use something generic instead:

Individual name: IOOS Data Manager
Phone: 240-533-9444
Email: data.ioos@noaa.gov

Thanks!

Comment (srstsavage) [reply docker-compose.yml]: Done!

File: docker-compose.yml [hunk 2]
```diff
@@ -1,18 +1,38 @@
-version: '2'
 services:
   erddap:
-    container_name: "erddap_gold_standard"
     image: axiom/docker-erddap:2.23-jdk17-openjdk
     restart: unless-stopped
     ports:
       - "8080:8080"
+    volumes:
+      - ./erddap/conf/robots.txt:/usr/local/tomcat/webapps/ROOT/robots.txt
```
Comment (MathewBiddle) [inline docker-compose.yml:8]: Don't you want to map from `%{PWD}/erddap/conf/robots.txt` ?

Comment (MathewBiddle) [reply docker-compose.yml:8]: similarly with the next few lines.

File: docker-compose.yml [hunk 3]
```diff
@@ -1,18 +1,38 @@
...
+      - ./erddap/content:/usr/local/tomcat/content/erddap
+      - ./erddap/data:/erddapData
+      - ./datasets:/datasets
     environment:
+      #note: these can also be set in /usr/local/tomcat/bin/config.sh
```
Comment (MathewBiddle) [inline docker-compose.yml:13]: The documentation should be updated to reflect this change.
https://github.com/ioos/erddap-gold-standard/blob/gh-pages/_docs/customize-erddap.md#update-configsh

File: docker-compose.yml (hunk 2 above)
Comment (srstsavage) [reply docker-compose.yml:8]: I don't think so, you can use relative paths in `docker-compose.yml` if they begin with `./` (unless you mean something else!)

**Pull request merged**